source/particle/generator/uniform_box.cc

```diff
@@ -0,0 +1,187 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/uniform_box.h>
+
+#include <boost/random.hpp>
+#include <deal.II/base/std_cxx11/array.h>
+
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      UniformBox<dim>::UniformBox() {}
+
+      template <int dim>
+      std::multimap<types::LevelInd, Particle<dim> >
+      UniformBox<dim>::generate_particles()
+      {
+        std::multimap<types::LevelInd, Particle<dim> > particles;
+
+        types::particle_index cur_id = 0;
+        const Tensor<1,dim> P_diff = P_max - P_min;
+
+        double volume(1.0);
+        for (unsigned int i = 0; i < dim; ++i)
+          volume *= P_diff[i];
+
+        std_cxx11::array<double,dim> nParticles;
```

Comment:
call it `n_particles_per_direction` or similar. Do you intend to create more than 4B particles in each direction?